rclpy/src/rclpy/_rclpy.c

```diff
@@ -2319,8 +2319,9 @@ rclpy_get_node_names(PyObject * Py_UNUSED(self), PyObject * args)
   PyObject * pynode_names = PyList_New(node_names.size);
   size_t idx;
   for (idx = 0; idx < node_names.size; ++idx) {
+    char * name = node_names.data[idx];
     PyList_SetItem(
-      pynode_names, idx, PyUnicode_FromString(node_names.data[idx]));
+      pynode_names, idx, name ? PyUnicode_FromString(name) : Py_None);
```

Comment:
I think `Py_INCREF(Py_None)` [is needed here](https://docs.python.org/3/c-api/none.html#c.Py_None) since `PyList_SetItem` steals a reference.

Reply:
Thank you for catching this (again). Fixed in 9135c1c865dc2c96063749dfd14fd3fbd9bd07b1. Please feel free to point out the third flaw :wink: